### PR TITLE
Add evm function call

### PR DESCRIPTION
**Pull request by Pessina**
This PR includes: 

- Add support for EVM Function Call using ChainSignatures
- Add local EVM blockchain using `hardhat` for testing purpose
- Fix Webkit bug on Submit button

Obs: The tests for this PR are expected to be flaky as they depend on relayer and ChainSignature contract that fail frequently. The solution would be run those dependencies locally (I can setup this later)

### CODE REVIEW

**Comment by Pessina** [comment]
@race-of-sloths

**Comment by race-of-sloths** [comment]
@Pessina Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
New Sloth joined the Race! Welcome!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/Pessina?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/Pessina?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/Pessina)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/Pessina)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: packages/near-fast-auth-signer-e2e-tests/contracts/FT.sol
Comment (hcho112) [inline packages/near-fast-auth-signer-e2e-tests/contracts/FT.sol:1]: so this file is extended version of file from above link? could you add a bit more detail about what has modified/extended and why? (eg. I notice that `mint` doesn't exist on above, but added below)

File: packages/near-fast-auth-signer/src/components/SignMultichain/SignMultichain.tsx
Comment (hcho112) [inline packages/near-fast-auth-signer/src/components/SignMultichain/SignMultichain.tsx:103]: is it ok to exclude `isUserAuthenticated` check?

**Review by hcho112 — COMMENTED** [review]
This is amazing work. Left few comments. Would prefer to get more reviews from rest of the team

File: packages/near-fast-auth-signer/src/components/SignMultichain/SignMultichain.tsx
Comment (Pessina) [reply packages/near-fast-auth-signer/src/components/SignMultichain/SignMultichain.tsx:103]: I moved it to here: https://github.com/near/fast-auth-signer/blob/db527505aabb414eb9443d389a979a20cd647872/packages/near-fast-auth-signer/src/components/SignMultichain/SignMultichain.tsx#L130

We need to be authenticated before estimate the gas for the function call.

File: packages/near-fast-auth-signer-e2e-tests/contracts/FT.sol
Comment (Pessina) [reply packages/near-fast-auth-signer-e2e-tests/contracts/FT.sol:1]: I didn't added any explanation because we will never change this file again. It's just a contract implementing the ERC20 interface for testing. 

The code here was copy pasted from the URL linked, and the mint function was added so we can distribute the tokens as we want for testing. 

I was checking other options to implement this contracts, looks like Open Zeppelin haves a npm package with the implementations for all we need ERC20, ERC721, ERC1155. 

But if it's ok I would like to tackle this in another PR, there is too much work here already. 

That's the package I'm looking to use: https://www.npmjs.com/package/@openzeppelin/contracts

**Comment by race-of-sloths** [comment]
### ❌ PR inactive
This PR has been inactive for two weeks. It's now marked as stale. 
>[!TIP]
If you think it's a mistake, you can restart the bot with `include` command to proceed 🔄

File: packages/near-fast-auth-signer/src/components/SignMultichain/bitcoin/schema.ts
Comment (esaminu) [inline packages/near-fast-auth-signer/src/components/SignMultichain/bitcoin/schema.ts:9]: Agree with replacing utxos and fees with inputs and outputs. what do you think about slightly more validation for inputs to help devs catch errors early?

```typescript
inputs: yup.array(yup.object({
  txid: yup.string().required(),
  vout: yup.number().required(),
  amount: yup.number().required()
})).optional()
```

File: packages/near-fast-auth-signer/src/components/SignMultichain/evm/schema.ts
Comment (esaminu) [inline packages/near-fast-auth-signer/src/components/SignMultichain/evm/schema.ts:5]: We should also validate the `chainId` like before. Also are we no longer supporting dapps specifying the gas themselves and only fetching when it's not specified?

File: packages/near-fast-auth-signer/src/components/SignMultichain/bitcoin/schema.ts
Comment (esaminu) [inline packages/near-fast-auth-signer/src/components/SignMultichain/bitcoin/schema.ts:6]: We should validate that a `from` is provided here. As discussed the dapp can choose to be explicit about the from and if it is provided the signer should check that it matches the derivation path

File: packages/near-fast-auth-signer/src/components/SignMultichain/evm/schema.ts
Comment (esaminu) [inline packages/near-fast-auth-signer/src/components/SignMultichain/evm/schema.ts:6]: Since it's no longer validated in the base schema, we should make sure that the dev has passed in a `to`, `value` and optionally a `from` here

File: packages/near-fast-auth-signer/src/components/SignMultichain/bitcoin/schema.ts
Comment (Pessina) [reply packages/near-fast-auth-signer/src/components/SignMultichain/bitcoin/schema.ts:6]: The BTC doesn't accept a from, even if provided it will be ignored, now and before. 

The EVM can accept a from and as discussed it will be validated on the multichain-tools, here is the place where the validation occurs: https://github.com/near/multichain-tools/blob/cce661c13a53b8b89dde5f3c82f3aadd1790a0aa/src/chains/EVM/EVM.ts#L156

File: packages/near-fast-auth-signer/src/components/SignMultichain/evm/schema.ts
Comment (Pessina) [reply packages/near-fast-auth-signer/src/components/SignMultichain/evm/schema.ts:5]: It's being validate on the derivationPath

File: packages/near-fast-auth-signer/src/components/SignMultichain/bitcoin/schema.ts
Comment (Pessina) [reply packages/near-fast-auth-signer/src/components/SignMultichain/bitcoin/schema.ts:9]: Good point, I will narrow the validation

File: packages/near-fast-auth-signer/src/components/SignMultichain/evm/schema.ts
Comment (Pessina) [reply packages/near-fast-auth-signer/src/components/SignMultichain/evm/schema.ts:6]: Good point, it will be fixed when we narrow down the schema

File: packages/near-fast-auth-signer/src/components/SignMultichain/evm/schema.ts
Comment (esaminu) [reply packages/near-fast-auth-signer/src/components/SignMultichain/evm/schema.ts:5]: What about EVM chain previously validated here:
```typescript
chainId: yup.lazy((value) => yup.mixed<bigint>().test(
    'is-valid-chainId',
    `Invalid chainId value: ${value}`,
    (val) => [BigInt(1), BigInt(56), BigInt(97),
```

**Review by esaminu — APPROVED** [review]
LGTM. Great work @Pessina !

**Pull request merged**
